spring-cloud-task-core/src/main/java/org/springframework/cloud/task/listener/annotation/TaskListenerExecutor.java

```
@@ -0,0 +1,246 @@
+				failedTaskMethods.keySet(),failedTaskInstances);
+	}
+
+	private void executeTaskListener(TaskExecution taskExecution, Set<Method> methods, Map<Method, Object> instances){
+		for (Method method : methods) {
+			try {
+				method.invoke(instances.get(method),taskExecution);
+			}
+			catch (IllegalAccessException e) {
+				throw new IllegalStateException("@BeforeTask and @AfterTask annotated methods must be public.");
+			}
+			catch (InvocationTargetException e) {
+				throw new IllegalStateException("Failed to process @BeforeTask or @AfterTask" +
+						"annotation because: ", e);
+			}
+			catch (IllegalArgumentException e){
+				throw new IllegalStateException("taskExecution parameter is required for @BeforeTask and @AfterTask annotated methods");
+			}
+		}
+	}
+
+	private void executeTaskListenerWithThrowable(TaskExecution taskExecution,
+			Throwable throwable, Set<Method> methods, Map<Method, Object> instances){
+		for (Method method : methods) {
+			try {
+				method.invoke(instances.get(method),taskExecution, throwable);
+			}
+			catch (IllegalAccessException e) {
+				throw new IllegalStateException("@FailedTask annotated methods must be public.");
+			}
+			catch (InvocationTargetException e) {
+				throw new IllegalStateException("Failed to process @FailedTask " +
+						"annotation because: ", e);
+			}
+			catch (IllegalArgumentException e){
+				throw new IllegalStateException("taskExecution and throwable parameters "
+						+ "are required for @FailedTask annotated methods");
+			}
+		}
+	}
+
+	private void initializeExecutor( ) {
+		ConfigurableListableBeanFactory factory = context.getBeanFactory();
+		for( String beanName : context.getBeanNamesForType(Object.class)) {
+
+			if (!ScopedProxyUtils.isScopedTarget(beanName)) {
+				Class<?> type = null;
+				try {
+					type = AutoProxyUtils.determineTargetClass(this.context.getBeanFactory(), beanName);
+				}
+				catch (Throwable ex) {
```

Swallowing `Throwable`?